core/vm/actor/builtin/types/miner/v0/monies.hpp

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <vm/actor/builtin/types/miner/policy.hpp>
+#include <vm/runtime/runtime.hpp>
+#include "vm/actor/builtin/types/miner/monies.hpp"
+#include "primitives/go/math.hpp"
+
+namespace fc::vm::actor::builtin::v0::miner {
+  using common::math::kPrecision128;
+  using common::smoothing::FilterEstimate;
+  using primitives::BigInt;
+  using primitives::ChainEpoch;
+  using primitives::StoragePower;
+  using primitives::TokenAmount;
+  using states::MinerActorState;
+  using states::Universal;
+  using types::miner::VestSpec;
+  using version::NetworkVersion;
+
+  class Monies : public types::miner::Monies {
+   private:
+    const BigInt lock_target_factor_num = BigInt(3);
+    const BigInt lock_target_factor_denom = BigInt(10);
+
+    const BigInt space_race_initial_pledge_max_per_byte =
+                     bigdiv(BigInt("1000000000000000000"), (BigInt{32} << 30));
+
+    constexpr static int declared_fault_factor_num_v0 = 214;
+    constexpr static int declared_fault_factor_num_v3 = 351;
+    constexpr static int declared_fault_factor_denom = 100;
+    const ChainEpoch declared_fault_projection_period_v0 =
+        ChainEpoch(kEpochsInDay * declared_fault_factor_num_v0
+                   / declared_fault_factor_denom);
+    const ChainEpoch declared_fault_projection_period_v3 =
+        ChainEpoch(kEpochsInDay * declared_fault_factor_num_v3
+                   / declared_fault_factor_denom);
+
+    constexpr static int undeclared_fault_factor_num_v0 = 50;
+    constexpr static int undeclared_fault_factor_num_v1 = 35;
+    constexpr static int undeclared_fault_factor_denom = 10;
+
+    const ChainEpoch undeclared_fault_projection_period_v0 =
+        ChainEpoch(kEpochsInDay * undeclared_fault_factor_num_v0
+                   / undeclared_fault_factor_denom);
+    const ChainEpoch undeclared_fault_projection_period_v1 =
+        ChainEpoch(kEpochsInDay * undeclared_fault_factor_num_v1
+                   / undeclared_fault_factor_denom);
+
+   public:
+    virtual outcome::result<TokenAmount> expectedRewardForPower(
+        const FilterEstimate &reward_estimate,
+        const FilterEstimate &network_power_estimate,
+        const StoragePower &sector_power,
+        const ChainEpoch &projection_duration) override;
```

Comment:
```suggestion
   outcome::result<TokenAmount> expectedRewardForPower(
        const FilterEstimate &reward_estimate,
        const FilterEstimate &network_power_estimate,
        const StoragePower &sector_power,
        const ChainEpoch &projection_duration) override;
```

Reply:
Use "override" methods without "virtual".
Use it for all methods

Reply:
fixed